boa/vm/py_evm.py

```diff
@@ -388,6 +388,12 @@ def enable_fast_mode(self, flag: bool = True):
             unpatch_pyevm_state_object(self.vm.state)
 
     def fork_rpc(self, rpc: RPC, block_identifier: str, **kwargs):
+        if len(self.vm.state._account_db._journaltrie._journal._current_values) > 0:
+            warnings.warn(
+                "forking with a dirty state, this is likely to cause issues",
+                stacklevel=2,
```

Comment:
How is this causing issues? we should start a totally new env by calling _init_vm.
Why not make it an error anyway? We should have helper functions so that we always start a fork with good state.

Reply:
Check out #182

Reply:
How about we make this error out by default and we add a `force` optional argument to `boa.env.fork`?

Reply:
Sounds good :ok_hand: 
Note the above PR wants to make this `boa.fork` instead of `boa.env.fork`. Always possible to `boa.env.fork(force=true)`

Reply:
I think forking a dirty env should be possible.

Also with https://github.com/vyperlang/titanoboa/pull/285, boa.fork creates a new env anyway, so we will need to change the detector. I think the issue is not really whether the env is dirty or not, but whether the user intended to fork at that point in the program